File: 1685-sum-of-absolute-differences-in-a-sorted-array/1685-sum-of-absolute-differences-in-a-sorted-array.cpp
class Solution {
public:
    vector<int> getSumAbsoluteDifferences(vector<int>& nums) {
        int n = nums.size();
        vector<int> prefix = {nums[0]};
        for(int i=1; i<n; i++) prefix.push_back(prefix[i-1]+nums[i]);
        vector<int> ans;
        for(int i=0; i<n; i++){
            int leftSum = prefix[i] - nums[i];
            int rightSum = prefix[n-1] - prefix[i];
            
            int leftCnt = i, rightCnt = n-1-i;
            
            int leftTotal = leftCnt*nums[i] - leftSum;
            int rightTotal = rightSum - rightCnt*nums[i];
            ans.push_back(leftTotal+rightTotal);
        }
        return ans;
    }
};

File: 1685-sum-of-absolute-differences-in-a-sorted-array/1685-sum-of-absolute-differences-in-a-sorted-array.py
class Solution {
    public int[] getSumAbsoluteDifferences(int[] nums) {
        int n = nums.length, totalSum = 0;
        for(int x : nums) totalSum += x;
        int leftSum = 0;
        int[] ans = new int[n];
        for(int i=0; i<n; i++){
            int rightSum = totalSum - leftSum - nums[i];
            
            int rightCnt = n-1-i, leftCnt = i;
            
            int leftTotal = leftCnt*nums[i] - leftSum;
            int rightTotal = rightSum - rightCnt*nums[i];
            
            ans[i] = leftTotal + rightTotal;
            leftSum += nums[i];
        }
        return ans;
    }
}